File: Builder/PHP/Director.php
<?php
  class Director {
    private $builder;

    public function __construct($builder) {
      $this->builder = $builder;
    }

    public function construct($header, $body, $footer) {
      $this->builder->buildHeader($header);
      $this->builder->buildBody($body);
      $this->builder->buildFooter($footer);
    }
  }

File: Builder/Go/builder/Director.go
package builder

type Director struct {
  builder builder
}

func NewDirector(builder builder) *Director {
  return &Director{builder: builder}
}

func (this *Director) Construct(header string, body string, footer string) {
  this.builder.buildHeader(header)
  this.builder.buildBody(body)
  this.builder.buildFooter(footer)
}
